src/vault.py

```diff
@@ -1,114 +1,158 @@
-                role_id=self.vault_approle_id,
-                secret_id=self.vault_approle_secret_id
+            vault_approle_auth = self.vault_object.auth.approle.login(
+                role_id=self.approle_id,
+                secret_id=self.secret_id
             )['auth']
-
-            # If the vault configuration is different from the default configuration
-            if self.vault_mount_point != "secretv2":
-                vault_object.secrets.kv.v2.configure(
-                    max_versions=5000,
-                    mount_point=self.vault_mount_point,
-                    cas_required=False,
-                )
-
             log.info(
                 f"[class.{__class__.__name__}] "
-                f"kv engine is configured, "
-                f"token id {vault_approle_auth['entity_id']}"
+                f"Vault Token {vault_approle_auth['entity_id']} created successful"
+            )
+        except hvac.exceptions.Forbidden as forbidden:
+            log.error(
+                f"[class.{__class__.__name__}] "
+                f"logging with approle forbidden, "
+                f"please check capabilities ['read', 'list', 'create', 'update] "
+                f"for '{mount_point}/config'\n"
+                f"{forbidden}"
+            )
+        except hvac.exceptions.InvalidRequest as invalidrequest:
+            log.error(
+                f"[class.{__class__.__name__}] "
+                f"logging with approle invalid request, "
+                f"please check your role-id or secret-id\n"
+                f"{invalidrequest}"
             )
-            self.vault_object = vault_object
-
-        except Exception as ex:
-            log.error(f"[class.{__class__.__name__}] exception init kv engine: {ex}")
-
 
-    def vault_read_secrets(self, path, secret_key=None):
+    def vault_read_secrets(
+            self,
+            path: str = None,
+            key: str = None
+    ):
+        """Function for read secrets from Vault.
+        :param path: The path to the secret in vault.
+        :type path: str
+        :default path: None
+        :param key: The key from which you want to read the value.
+        :type key: str
+        :default key: None
+        """
         try:
             read_response = self.vault_object.secrets.kv.v2.read_secret_version(
                                     path=path,
-                                    mount_point=self.vault_mount_point,
+                                    mount_point=self.mount_point,
             )
-
-            if secret_key is None:
+            if key is None:
                 response = read_response['data']['data']
             else:
-                response = read_response['data']['data'][secret_key]
-
+                response = read_response['data']['data'][key]
         except hvac.exceptions.InvalidPath:
             read_response = {'data': {'data': {'exception': 'InvalidPath'}}}
             response = read_response['data']['data']
```

Comment:
There are doubts about this stub.
This is an old artifact and I used it to solve the exception problem for one specific project.
Since the module is universal and used in different projects, it makes sense to remove this artifact and solve the problem on the side of a specific project.
We need to check why it was used...